### PR TITLE
Add a resolveModuleId and resolveModuleImport callback to options.

### PR DESCRIPTION
**Pull request by ritzingerp**
With this callback callers can provide a custom logic how the identifiers of referenced modules will be written into the generated d.ts

### CODE REVIEW

**Comment by dylans** [comment]
Thanks @ritzingerp . I think this looks fine. Will let it sit a little longer in case anyone else wants to review. Will land it for 2.1.0.

**Comment by dylans** [comment]
@ritzingerp resolving #90 has introduced a small conflict here... please rebase and then I'll land this. Also, if you have time, please also add a PR to update the README. Thanks!

**Comment by ritzingerp** [comment]
The rebase is done. I have extended this pull request to not only be able to customize module ids in `import`, but also in `declare module` statements.

I have also created a unit test case.

File: index.ts [hunk 1]
```diff
@@ -56,11 +74,11 @@ function getError(diagnostics: ts.Diagnostic[]) {
 		// the tsconfig.json don't; the messageText is enough to diagnose in those
 		// cases.
 		if (diagnostic.file) {
-			const position = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
+		const position = diagnostic.file.getLineAndCharacterOfPosition(diagnostic.start);
```
Comment (dylans) [inline index.ts]: Looks like the indentation here is off by one?

File: index.ts [hunk 2]
```diff
@@ -56,11 +74,11 @@ function getError(diagnostics: ts.Diagnostic[]) {
...
 
-			message +=
-				`\n${diagnostic.file.fileName}(${position.line + 1},${position.character + 1}): ` +
-				`error TS${diagnostic.code}: ${diagnostic.messageText}`;
+		message +=
```
Comment (dylans) [inline index.ts]: indentation change unintentional?

**Comment by ritzingerp** [comment]
I have fixed the indentation, it was not intended.

**Comment by ritzingerp** [comment]
Originally this pull request introduced a breaking change.

For input file `some/module.ts`:
```ts
import { x } from 'x';
```

the following was generated before the pull request:
```ts
declare module 'name/some/module' {
    import { x } from 'x';
}
```

and originally with the pull request:
```ts
declare module 'name/some/module' {
    import { x } from 'name/x'; // 'name/' prepended to import, only if there was no declare module 'x' in input
}
```

The latest commit for this pull request reverses this behavior, and does not introduce a breaking change.

This might be related to #94, but I think this pull request should not include a breaking change.

File: tests/support/foo-resolve-module-id/NonRel.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+﻿export class NonRel {
```
Comment (dylans) [inline tests/support/foo-resolve-module-id/NonRel.ts]: I'm going to be picky here and say that we should call this NonRelative throughout, given that Rel can have multiple meanings.

**Comment by ritzingerp** [comment]
Agree, renamed to NonRelative.

**Comment by dylans** [comment]
Thanks @ritzingerp !

**Pull request merged**